modules/fargate/locals.tf

```diff
@@ -0,0 +1,10 @@
+locals {
+  create_eks              = var.create_eks && length(var.fargate_profiles) > 0
```

Comment:
I think it is better to introduce another variable that should be known in advance (eg, type boolean) to be able to use it in `count`. Something like `create_eks_fargate = true`.

The way it is written now `fargate_profiles` can't contain references to not-yet-created resources and modules inside of the value.